models/packages/package.go

```diff
@@ -190,7 +190,7 @@ func GetPackagesByType(ctx context.Context, ownerID int64, packageType Type) ([]
 // DeletePackagesIfUnreferenced deletes a package if there are no associated versions
 func DeletePackagesIfUnreferenced(ctx context.Context) error {
 	in := builder.
-		Select("package_version.package_id").
+		Select("package.id").
```

Comment:
This is enough
```go
in := builder.Select("package_version.package_id").
From("package_version").
Where(builder.Expr("package_version.id IS NULL"))
```

Reply:
No, `package_version.id` is never `NULL`. That works only because of the left join. Sadly MySql is the only database which does not support subqueries with the table which should be modified.

https://stackoverflow.com/questions/4471277/mysql-delete-from-with-subquery-as-condition